extensions-core/hdfs-storage/src/main/java/org/apache/druid/inputsource/hdfs/HdfsInputSource.java

```diff
@@ -214,10 +215,23 @@ public boolean needsFormat()
   private void cachePathsIfNeeded() throws IOException
   {
     if (cachedPaths == null) {
-      cachedPaths = ImmutableList.copyOf(Preconditions.checkNotNull(getPaths(inputPaths, configuration), "paths"));
+      Collection<Path> paths = Preconditions.checkNotNull(getPaths(inputPaths, configuration), "paths");
+      cachedPaths = ImmutableList.copyOf(paths);
     }
   }
 
+  @Override
+  public void validateAllowDenyPrefixList(InputSourceSecurityConfig securityConfig)
+  {
+    try {
+      cachePathsIfNeeded();
```

Comment:
I think you won't probably need to cache paths since this will unnecessarily populate the cache in the Overlord.

Reply:
done.